src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitRepositoryFSManager.java

```diff
@@ -0,0 +1,245 @@
+package de.uhd.ifi.se.decision.management.jira.extraction.versioncontrol;
+
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+
+public class GitRepositoryFSManager {
+	private static final String TEMP_DIR_PREFIX = "TEMP";
+	private static final long BRANCH_OUTDATED_AFTER = 60 * 60 * 1000; //ex. 1 day = 24 hours * 60 minutes * 60 seconds * 1000 miliseconds
+	private String basePath;
+	private String baseProjectPath;
+	private String baseProjectUriPath;
+	private String baseProjectUriDefaultPath;
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitRepositoryFSManager.class);
+
+	public GitRepositoryFSManager(String home, String project, String repoUri, String defaultBranch) {
+		basePath = home;
+		baseProjectPath = basePath + File.separator + project;
+		baseProjectUriPath = baseProjectPath + File.separator + getShortHash(repoUri);
+		baseProjectUriDefaultPath = baseProjectUriPath + File.separator + defaultBranch;
+		// clean up if possible after previous requests.
+		maintainNotUsedBranchPaths();
+	}
+
+	/**
+	 * Returns target directory path for the default branch of the repository.
+	 * @return absolute path to directory of the default branch
+	 */
+	public String getDefaultBranchPath() {
+		return baseProjectUriDefaultPath;
+	}
+
+	/**
+	 * Makes branch's folder available in the temporary pool.
+	 * It can significantly contribute to improving the speed
+	 * of check-outs of other branches as folder renaming
+	 * is not costly compared to copying.
+	 * 
+	 * This mehod stays public as the developer might intend
+	 * to release the folder and not wait for maintenance
+	 * strategy to trigger it.
+	 *
+	 * @param branchShortName branch name
+	 * @return null on failure, absolute path to new temporary directory
+	 */
+	public String releaseBranchDirectoryNameToTemp(String branchShortName) {
+		File oldDir = new File(getBranchPath(branchShortName));
+		if (!oldDir.isDirectory()) {
+			return null;
+		}
+		else {
+			Date date= new Date();
+			long time = date.getTime();
+			String tempDirString = baseProjectUriPath
+					+File.separator
+					+TEMP_DIR_PREFIX+String.valueOf(time);
+			File tempDir = new File(tempDirString);
+			try {
+                /* TODO: An issue is observable on Windows machine.
```

Comment:
An issue is "observed"? You might indicate the issue by the @issue tag in the comment.